src/compiler/compile.js

```diff
@@ -588,12 +590,31 @@ function checkTerminalDirectives (el, options) {
       return skip
     }
   }
-  var value, dirName
-  for (var i = 0, l = terminalDirectives.length; i < l; i++) {
-    dirName = terminalDirectives[i]
-    value = el.getAttribute('v-' + dirName)
-    if (value != null) {
-      return makeTerminalNodeLinkFn(el, dirName, value, options)
+
+  var terminals = terminalDirectives.map(function (terminal) {
```

Comment:
These change seems pretty expensive since it's now map + sort + nested for loop for every single element compiled. Ideally we want to move as much work out of this function as possible.